pages/api/workshop.ts

```diff
@@ -0,0 +1,64 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { table, base } from "../../utils/table"
+import IEvent from '../../utils/IEvent'
+
+const filterStrings = [
+  'Test Event',
+  'The Ultimate Future Event',
+  'Hack Night'
+]
+
+const handler = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  const { method } = req;
+  switch (method) {
+    case 'GET':
+      const events:IEvent[] = [];
+      table.select({
+      }).eachPage(function page(records, fetchNextPage) {
+        
+        RecordsLoop:
+        for (const record of records) {
+          const eventName = record.fields['Event Name'] as string
+          const eventDateStr = record.fields['Event Date & Start Time'] as string
+          const eventDate = new Date(eventDateStr)
+          const recapImg = record.fields['Recap Images'] ?? []
+
+          // maybe there is a better way to filter out events
+          for (const string of filterStrings) {
+            if (eventName.substring(0, string.length) == string) {
+              continue RecordsLoop;
+            }
+          }
+
+          // only display events with images
+          if (recapImg[0] === undefined) continue
```

Comment:
You can also include this directly in `filterByFormula`. I don't remember exactly the syntax but you can use an Airtable formula to check if an attachment field is not empty.